### PR TITLE
Swift 6-ify the package manifest

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

v2 will require the Swift 6 language mode. In order to continue developing v1 and v2 on main we'll need a Swift 6 specific manifest version in addition to the 5.x version.

Moving forward we'll have two package manifests, `Package.swift` for 5.x containing only v1, and `Package@swift-6.swift` using tools version 6.0 targetting v1 and v2.

Modifications:

- Remove v2 targets from Package.swift
- Add a second manifest and make it warning free with Swift 6, this requires turning `static let`s to `static var`s.
- Conditionalise some imports in `protoc-gen-grpc-swift` (as it is shared by v1 and v2)
- Fix an error in the plugin for Swift 6

Result:

v2 now only compiles with a Swift 6 compiler

### CODE REVIEW

File: Package@swift-6.swift [hunk 1]
```diff
@@ -0,0 +1,927 @@
...
+
+  static var grpcCore: Target {
+    .target(
+      name: "GRPCCore",
+      dependencies: [
+        .dequeModule,
+        .atomics
+      ],
+      path: "Sources/GRPCCore",
+      swiftSettings: [.swiftLanguageVersion(.v5)]
```
Comment (gjcairo) [inline Package@swift-6.swift:196]: Do we want to adopt Swift 6 language version for the v2 packages? Or not yet?

File: Sources/protoc-gen-grpc-swift/main.swift [hunk 2]
```diff
@@ -169,6 +172,7 @@ func main(args: [String]) throws {
           generatedFiles: &generatedFiles
         )
         if options.v2 {
+          #if compiler(>=6.0)
```
Comment (gjcairo) [inline Sources/protoc-gen-grpc-swift/main.swift:175]: Should we `fatalError` (or error in some other way) if the compiler is not Swift 6+ but the v2 option is set?

File: Package@swift-6.swift (hunk 1 above)
Comment (glbrntt) [reply Package@swift-6.swift:196]: I'm going to make the changes in a separate PR to keep the diff small(er). This way we can do it module by module (or at least do the larger ones separately)

File: Sources/protoc-gen-grpc-swift/main.swift (hunk 2 above)
Comment (glbrntt) [reply Sources/protoc-gen-grpc-swift/main.swift:175]: Hmm, good question. I think we should actually put the v2 option behind the compiler guard too tbh.

**Review by gjcairo — APPROVED** [review]

**Pull request merged**